### PR TITLE
modify add_cluster.go and add new examples

### PR DESCRIPTION
**Pull request by huihui0311**
### Description of the Change

- Correct a syntax error in  add_cluster.go

- Add new examples: 
add_fence_agent.go
add_iscsi_data_storage_domain.go
add_openstack_image_provider.go
add_role.go
add_vm_pool.go
assign_permission.go
assign_permission_to_vms.go
asynchronous_inventory.go
change_host_cluster.go
data_center_maintenance.go
list_tags.go
deactivate_disk_attachment.go
disable_compression.go
download_vm_ovf.go
export_template.go
export_template_as_ova.go
export_vm_as_ova.go
extend_disk.go
get_display_ticket.go
get_vm_ip.go
import_vm_from_ova.go
import_vm_from_vmware.go
page_vms.go
poll_events.go
print_vm_mac_duplicates.go
remove_numa_nodes_from_vm.go
set_vm_serial_number.go
show_summary.go
sparsify_disk.go
start_vm_with_boot_devices.go
test_connection.go
upgrade_host.go
vm_backup.go

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: sdk/examples/add_iscsi_data_storage_domain.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		}
+	}()
+
+	// use add method to create storageDomain
+	// Note that 'myhost' should be in a dc with version >= 4.1 for enabling
+	// the 'Discard After Delete' (DAD) parameter, since only from that
+	// version, the host returns the luns' discard related information.
+	storageDomainService := conn.SystemService().StorageDomainsService()
+
+	logicalUnit := &ovirtsdk4.LogicalUnit{}
```
Comment (imjoey) [inline sdk/examples/add_iscsi_data_storage_domain.go]: Please use `LogicalUnitBuilder` instead. Thanks.

File: sdk/examples/add_openstack_image_provider.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+				ovirtsdk4.NewOpenStackImageProviderBuilder().
+					Name("myglance").
+					Description("My Glance").
+					Url("http://glance.ovirt.org:9292").
+					RequiresAuthentication(false).
+					MustBuild()).
+			Send()
+	}
+
+	// Note that the provider that we are using in this example is public
```
Comment (imjoey) [inline sdk/examples/add_openstack_image_provider.go]: Shall we remove the unused code below?

File: sdk/examples/asynchronous_inventory.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	// way the requests will be sent simultaneously, using the multipl connections
+	systemService := conn.SystemService()
+	fmt.Printf("Requesting data...")
+	dcsListFuture, err := systemService.DataCentersService().List().Send()
+	clustersListFuture, err := systemService.ClustersService().List().Send()
+	sdsListFuture, err := systemService.StorageDomainsService().List().Send()
+	networksListFuture, err := systemService.NetworksService().List().Send()
+	hostsListFuture, err := systemService.HostsService().List().Send()
+	vmsListFuture, err := systemService.VmsService().List().Send()
+	disksListFuture, err := systemService.DisksService().List().Send()
```
Comment (imjoey) [inline sdk/examples/asynchronous_inventory.go]: If the returned `err` is ignored, we should use `_` instead.

File: sdk/examples/add_openstack_image_provider.go (hunk 2 above)
Comment (huihui0311) [reply sdk/examples/add_openstack_image_provider.go]: This is an another demo of add openstack image provider which need additional security attributes.
It is necessary to keep it.

Comment (imjoey) [reply sdk/examples/add_openstack_image_provider.go]: Yes, it's definitely should be kept. My apologize.

File: sdk/examples/add_fence_agent.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	}
+	hostSlice, _ := listResp.Hosts()
+	host := hostSlice.Slice()[0]
+
+	// Find the service that manages the hosts
+	hostService := hostsService.HostService(host.MustId())
+
+	fenceAgentsService := hostService.FenceAgentsService()
+	// Use the "add" method to create a fence agents:
+	option := &ovirtsdk4.Option{}
```
[the remark this answers is not in the document]
Comment (huihui0311) [reply sdk/examples/add_fence_agent.go]: The update has been completed, thanks for your review.

File: sdk/examples/add_iscsi_data_storage_domain.go (hunk 1 above)
Comment (huihui0311) [reply sdk/examples/add_iscsi_data_storage_domain.go]: The update has been completed, thanks for your review.

File: sdk/examples/asynchronous_inventory.go (hunk 3 above)
Comment (huihui0311) [reply sdk/examples/asynchronous_inventory.go]: The update has been completed, thanks for your review.

**Comment by imjoey** [comment]
Hi @huihui0311 , could you please squash the commits? Thank you.

**Comment by imjoey** [comment]
@huihui0311  Thank you so much for your great contributions. Your work is definitely a tremendous amount of work. Let's merge them, thank again. 😄

**Comment by imjoey** [comment]
@huihui0311 Thank you so much for your great contributions. Your work is definitely a tremendous amount of work. Let's merge them, thank again. 😄

**Pull request merged**